CMakeLists.txt

```diff
@@ -325,9 +325,13 @@ if(AVIF_ENABLE_NODISCARD)
     else()
         unset(CMAKE_C_STANDARD)
         set_property(TARGET avif_obj PROPERTY C_STANDARD)
-        target_compile_options(avif_obj PUBLIC $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:C>:-std=gnu2x>>)
+        if(CMAKE_C_COMPILER_ID MATCHES "Clang" OR CMAKE_C_COMPILER_ID MATCHES "GNU")
+            target_compile_options(avif_obj PUBLIC $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:C>:-std=gnu2x>>)
```

Comment:
What needed `-std=gnu2x` as opposed to C++14/17?

Reply:
This is the C standard,  not the C++ standard. The `[[...]]` attribute syntax was added in C23, so `[[nodiscard]]` requires C23. The `-std=gnu2x` option is intended to request C23.

Note: In C++, The `[[...]]` attribute syntax was added in C++11, and the `nodiscard` attribute was added in C++17, so `[[nodiscard]]` requires C++17.

I just realized that we should use the equivalent [`/std:clatest`](https://learn.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=msvc-170) option for MSVC. Please see the second commit.

Reply:
Thanks, I didn't look at the context closely enough and assumed it was c++ related.